internal/service/domain/tree/import_tree_test.go

```diff
@@ -0,0 +1,193 @@
+package tree
+
+import (
+	"context"
+	"github.com/green-ecolution/green-ecolution-backend/internal/entities"
+	service "github.com/green-ecolution/green-ecolution-backend/internal/service/_mock"
+	storageMock "github.com/green-ecolution/green-ecolution-backend/internal/storage/_mock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"testing"
+	"time"
+)
+
+func TestTreeService_ImportTree(t *testing.T) {
+	ctx := context.Background()
+
+	treeToImport := &entities.TreeImport{
+		Latitude:     54.801539,
+		Longitude:    9.446741,
+		PlantingYear: 2023,
+		Species:      "Oak",
+		Number:       "T001",
+	}
+	t.Run("should create a new tree when no matching tree exists", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		imageRepo := storageMock.NewMockImageRepository(t)
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		expectedTree := getTestTrees()[0]
+
+		// Mock GetByCoordinates to simulate no tree found
+		treeRepo.EXPECT().GetByCoordinates(ctx, treeToImport.Latitude, treeToImport.Longitude).Return(nil, nil)
+
+		// Mock Create to confirm new tree is created
+		treeRepo.EXPECT().Create(ctx,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).Return(expectedTree, nil)
+
+		// When
+		err := svc.ImportTree(ctx, []*entities.TreeImport{treeToImport})
+
+		// Then
+		assert.NoError(t, err)
+	})
+
+	t.Run("should update an existing tree when matching coordinates and planting year are found", func(t *testing.T) {
```

Comment:
Maybe we can add a test case if the updating fails?

Reply:
I think we don't need to test the failure cases for update because we have already covered them in tree_test.go.